claripy/backends/backend_smtlib.py

```diff
@@ -0,0 +1,307 @@
+
+
+def _normalize_arguments(expr_left, expr_rigth):
+    """
+    Since we decided to emulate integer with bitvector, this method transform their
+    concrete value (if any) in the corresponding integer
+    """
+    if expr_left.is_str_op() and expr_rigth.is_bv_constant():
+        return expr_left, Int(expr_rigth.bv_signed_value())
+    elif expr_left.is_bv_constant() and expr_rigth.is_str_op():
+        return expr_rigth, Int(expr_left.bv_signed_value())
+    return expr_left, expr_rigth
+
+
+
+class BackendSMTLibBase(Backend):
+    def __init__(self, *args, **kwargs):
+        self.daggify = kwargs.pop('daggify', True)
+        self.reuse_z3_solver = False
+        Backend.__init__(self, *args, **kwargs)
+
+        # ------------------- LEAF OPERATIONS ------------------- 
+        self._op_expr['StringV'] = self.StringV
+        self._op_expr['StringS'] = self.StringS
+        self._op_expr['BoolV'] = self.BoolV
+        self._op_expr['BoolS'] = self.BoolS
+        self._op_expr['BVV'] = self.BVV
+        self._op_expr['BVS'] = self.BVS
+
+        # ------------------- BITVECTOR OPERATIONS -------------------
+        # self._op_raw['Extract'] = self._op_raw_extract
+        # self._op_raw['Concat'] = self._op_raw_concat
+        self._op_raw['__add__'] = self._op_raw_add
+        self._op_raw['__sub__'] = self._op_raw_sub
+
+        # ------------------- GENERAL PURPOSE OPERATIONS ------------------- 
+        self._op_raw['__eq__'] = self._op_raw_eq
+        self._op_raw['__ne__'] = self._op_raw_ne
+        self._op_raw['__lt__'] = self._op_raw_lt
+        self._op_raw['__le__'] = self._op_raw_le
+        self._op_raw['__gt__'] = self._op_raw_gt
+        self._op_raw['__ge__'] = self._op_raw_ge
+        self._op_raw['Or'] = self._op_raw_or
+        self._op_raw['If'] = self._op_raw_if
+        self._op_raw['Not'] = self._op_raw_not
+        self._op_raw['And'] = self._op_raw_and
+
+        # ------------------- STRINGS OPERATIONS ------------------- 
+        self._op_raw['StrConcat'] = self._op_raw_str_concat
+        self._op_raw['StrSubstr'] = self._op_raw_str_substr
+        self._op_raw['StrExtract'] = self._op_raw_str_extract
+        self._op_raw['StrLen'] = self._op_raw_str_strlen
+        self._op_raw['StrReplace'] = self._op_raw_str_replace
+        self._op_raw["StrContains"] = self._op_raw_str_contains
+        self._op_raw["StrPrefixOf"] = self._op_raw_str_prefixof
+        self._op_raw["StrSuffixOf"] = self._op_raw_str_suffixof
+        self._op_raw["StrIndexOf"] = self._op_raw_str_indexof
+        self._op_raw["StrToInt"] = self._op_raw_str_strtoint
+        self._op_raw["StrIsDigit"] = self._op_raw_isdigit
+        self._op_raw["IntToStr"] = self._op_raw_inttostr
+
+    @property
+    def is_smt_backend(self):
+        return True
+
+    def _smtlib_exprs(self, exprs):
+        return _exprs_to_smtlib(*exprs, daggify=self.daggify)
+
+    def _get_satisfiability_smt_script(self, constraints=(), variables=()):
+        '''
```

Comment:
All method-level comments should be wrapped around using triple double-quotes (`"""`) instead of single quotes. Can you please fix all comments in this PR?

Reply:
Fixed everywhere now